File: 05_Math,_Prime,_Greedy/9020_Goldbach’s_Conjecture/Answer9020.kt
fun main(){
  val n = readln().toInt()
  val inputNums = IntArray(n)
  for(i in 0 until n) inputNums[i] = readln().toInt()
  val primes = IntArray(10000){1}
  primes[0] = 0
  var i = 2
  while (i <= 10000/i){
    if(primes[i - 1] == 1){
      for(j in i .. 10000/i) primes[i * j - 1] = 0
    }
    i++
  }

  val resultStr = StringBuilder()
  for(h in 0 until n){
    var a = inputNums[h]/2
    var b = inputNums[h]/2
    if(h > 0) resultStr.append("\n")
    while(true){
      if(primes[a - 1] + primes[b - 1] == 2){
        resultStr.append("$a $b")
        break
      }
      a--
      b++
    }
  }

  print(resultStr)
}

File: 05_Math,_Prime,_Greedy/9020_Goldbach’s_Conjecture/Answer9020.ts
const [n, ...inputNums] = `${require('fs').readFileSync(0)}`.split('\n').map(v=>+v)
const primes = new Array(10000).fill(1)
primes[0] = 0;
for(let i = 2; i <= 10000/i; i++) {
  if(primes[i - 1]){
    for(let j = i; j <= 10000/i; j++) primes[i * j - 1] = 0
  }
}

const resultStr = []
for(let h = 0 ; h < n ; h++){
  let a = inputNums[h]/2;
  let b = inputNums[h]/2;
  while(true){
    if(primes[a - 1] + primes[b - 1] === 2){
      resultStr.push(`${a} ${b}`);
      break
    }
    a--
    b++
  }
}

console.log(resultStr.join('\n'))